clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -214,11 +215,51 @@ public String toString() {
         private final boolean shallow;
         private RecordsIterator innerIter;
 
+        // The variables for inner iterator
+        private final LinkedList<LogEntry> logEntries;
+        private final long lastInnerOffset;
+        private final long wrapperRecordOffset;
+
         public RecordsIterator(ByteBuffer buffer, CompressionType type, boolean shallow) {
             this.type = type;
             this.buffer = buffer;
             this.shallow = shallow;
             this.stream = Compressor.wrapForInput(new ByteBufferInputStream(this.buffer), type);
+            this.logEntries = null;
+            this.lastInnerOffset = -1L;
+            this.wrapperRecordOffset = -1L;
+        }
+
+        private RecordsIterator(LogEntry entry, CompressionType type, boolean shallow) {
```

Comment:
Since this is always called on the inner records, we probably don't need shallow in the param?